pkg/kp/consulutil/session_controller.go

```diff
@@ -65,6 +68,11 @@ func SessionManager(
 				sessionLogger.NoFields().Info("session manager: released session")
 			}
 			output <- ""
+
+			// Wait a random amount of time between 0 and 10 seconds before attempting to
+			// re-establish a session. This protects against a thundering herd of sessions
+			// attempting to be re-establish
```

Comment:
`rand.Seed(…)` needs to happen somewhere around here or this won't be all that random. `time.Now()` is a popular choice though of course it's not suitable to use here at this line of code because time.Now() might be the same for all callers!